### PR TITLE
Gazettes ingestor

### PR DESCRIPTION
**Pull request by actlikewill**
- Adds and ingestor for gazettes

### CODE REVIEW

**Comment by actlikewill** [comment]
@longhotsummer I am running into this issue when running this code. 
```
 raise ValueError('Cannot assign "%r": the current database router prevents this relation.' % value)
ValueError: Cannot assign "<Work: /akn/ke/officialGazette/government-gazette/1906-03-15/153 - Kenya Government Gazette dated 1906-03-15 number 153>": the current database router prevents this relation.
```

I am not sure why because I am not trying to create cross database relationships. I tried adding the minimal db router to but it is still not working.

**Comment by actlikewill** [comment]
@longhotsummer I'm having trouble figuring out how to copy the source file data  using the DynamicStorageFileField. :thinking:

File: peachjam/adapters/gazettes.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        super().__init__(settings)
+        self.jurisdiction = self.settings["jurisdiction"]
+
+    def check_for_updates(self, last_refreshed):
+        log.info(
+            f"Checking for new gazettes from Gazettes.Africa since {last_refreshed}"
+        )
+        if last_refreshed:
+            new_gazettes = (
+                CoreDocument.objects.filter(
```
Comment (longhotsummer) [inline peachjam/adapters/gazettes.py:26]: `doc_type="gazette"`

File: peachjam/adapters/gazettes.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+            }
+
+            if document.locality:
+                data["locality"] = Locality.objects.get(code=document.locality.code)
+
+            doc, new = Gazette.objects.update_or_create(
+                expression_frbr_uri=expression_frbr_uri, defaults={**data}
+            )
+
+            source_file = SourceFile(
```
Comment (longhotsummer) [inline peachjam/adapters/gazettes.py:77]: check for an existing source file

File: peachjam/adapters/gazettes.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+                document=doc,
+                filename=document.source_file.filename,
+                file=document.source_file.file,
```
Comment (longhotsummer) [inline peachjam/adapters/gazettes.py:80]: Here's a way to get the raw file field:

```python
>>> SourceFile.objects.filter(pk=1132).values_list('file', flat=True)[0]
's3:lawsafrica-gazettes-archive:archive/sc/2022/sc-government-gazette-dated-2022-10-21-no-67.pdf'
```

And you can write it directly with `update_or_create`

File: peachjam/adapters/gazettes.py (hunk 1 above)
Comment (actlikewill) [reply peachjam/adapters/gazettes.py]: Done

File: peachjam/adapters/gazettes.py (hunk 3 above)
Comment (actlikewill) [reply peachjam/adapters/gazettes.py]: @longhotsummer I have made this change but when the source file is saved it the `s3:lawsafrica-gazettes-archive` prefix  gets chopped in the database off even though I am copying the value directly as it is. Also I tried editing the value directly in the database to add the prefix but I still get the file not found error using the lawlibrary AWS credentials. Changing the AWS credentials to the gazette ones works.

File: peachjam/adapters/gazettes.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+            )
+
+            ga_source_file = (
+                SourceFile.objects.filter(document=ga_gazette)
+                .values("file", "filename", "size", "mimetype")
+                .using("gazettes_africa")
+            )
+
+            if ga_source_file:
+                SourceFile.objects.update_or_create(
```
Comment (longhotsummer) [inline peachjam/adapters/gazettes.py:78]: If this doesn't work then the final option is to use a raw sql query to update that field.

Comment (actlikewill) [reply peachjam/adapters/gazettes.py]: Added

File: peachjam/adapters/gazettes.py (hunk 3 above)
Comment (actlikewill) [reply peachjam/adapters/gazettes.py]: I'm curious whether the code you showed me on gazettemachine for updating source file actually works. Does it copy the source file path with the prefix?

**Comment by longhotsummer** [comment]
We need a way of testing this. ZA has 50,000+ gazettes which is crazy if it goes wrong. Instead, let's merge this and then set it up for new.ulii.org and test it with that, since uganda only has a few thousand gazettes.

**Comment by actlikewill** [comment]
> 

Got it.

**Pull request merged**